### PR TITLE
Added confirmation for reload/ESC when user has unsaved changes

### PR DESCRIPTION
**Pull request by vxsx**
When content editor has unsaved changes and tries to leave the page
or presses ESC, CMS will now ask if user really wants to do that.

Ref: #3349 

- [x] changelog
- [x] finalize messages /cc @evildmp 
- [x] tests

### CODE REVIEW

File: cms/templates/cms/toolbar/toolbar_javascript.html [hunk 1]
```diff
@@ -31,9 +31,10 @@
             'error': '<strong>{% filter escapejs %}{% trans "The following error occured:" %}{% endfilter %}</strong> ',
             'success': '{% filter escapejs %}{% trans "Action successfull... reloading." %}{% endfilter %}',
             'confirm': '{% filter escapejs %}{% trans "Are you sure you want to delete this plugin?" %}{% endfilter %}',
-            'confirmEmpty': '{% filter escapejs %}{% trans "({1}) plugin is empty. Remove it?" %}{% endfilter %}',
```
Comment (vxsx) [inline cms/templates/cms/toolbar/toolbar_javascript.html]: this was removed with sane-add-plugin, just cleaning up

**Comment by lory87** [comment]
lgtm

**Comment by FinalAngel** [comment]
LGTM, but where are the tests ;)

**Comment by lory87** [comment]
lgtm

**Comment by FinalAngel** [comment]
nice 👍

File: cms/templates/cms/toolbar/toolbar_javascript.html [hunk 2]
```diff
@@ -31,9 +31,10 @@
             'error': '<strong>{% filter escapejs %}{% trans "The following error occured:" %}{% endfilter %}</strong> ',
             'success': '{% filter escapejs %}{% trans "Action successfull... reloading." %}{% endfilter %}',
```
Comment (evildmp) [inline cms/templates/cms/toolbar/toolbar_javascript.html]: Spelling: *successful*

File: cms/templates/cms/toolbar/toolbar_javascript.html [hunk 3]
```diff
@@ -31,9 +31,10 @@
...
             'confirm': '{% filter escapejs %}{% trans "Are you sure you want to delete this plugin?" %}{% endfilter %}',
-            'confirmEmpty': '{% filter escapejs %}{% trans "({1}) plugin is empty. Remove it?" %}{% endfilter %}',
             'publish': '{% filter escapejs %}{% trans "Are you sure you want to publish this page?" %}{% endfilter %}',
-            'addPluginPlaceholder': '{% filter escapejs %}{% trans "Plugin will be added here" %}{% endfilter %}'
+            'addPluginPlaceholder': '{% filter escapejs %}{% trans "Plugin will be added here" %}{% endfilter %}',
+            'confirmDirty': '{% filter escapejs %}{% trans "There are some unsaved changes." %}{% endfilter %}',
```
Comment (evildmp) [inline cms/templates/cms/toolbar/toolbar_javascript.html]: _You have unsaved changes._

File: cms/templates/cms/toolbar/toolbar_javascript.html [hunk 4]
```diff
@@ -31,9 +31,10 @@
...
+            'confirmDirtyESC': '{% filter escapejs %}{% trans "Are you sure you want to stop editing?" %}{% endfilter %}'
```
Comment (evildmp) [inline cms/templates/cms/toolbar/toolbar_javascript.html]: Is it possible to change the text on the dialog buttons? That makes a difference to what we say here.

Comment (evildmp) [inline cms/templates/cms/toolbar/toolbar_javascript.html]: If we can't, then *Are you sure you want to abandon these changes?*

Comment (vxsx) [inline cms/templates/cms/toolbar/toolbar_javascript.html]: discussed this separately, not possible to affect buttons, result: "Are you sure you want to abandon these changes?"

**Comment by vxsx** [comment]
screen shot 2016-06-02 at 09 29 18

**Pull request merged**